sample_project/Assets/SampleViewer/Samples/Identify/Scripts/Identify.cs

```diff
@@ -0,0 +1,302 @@
+                ParseResults(SelectedResult, resultValue);
+
+                if (SelectedResult == 0)
+                {
+                    increaseResult.interactable = true;
+                    decreaseResult.interactable = false;
+                }
+                else
+                {
+                    DisableButtons(true);
+                }
+            }
+        });
+
+        Shader.SetGlobalFloat("_SelectedObjectID", 0);
+        Shader.SetGlobalColor("_HighlightColor", selectColor);
+    }
+
+    public void StartRaycast()
+    {
+#if !UNITY_IOS && !UNITY_ANDROID && !UNITY_VISIONOS
+        Ray ray = Camera.main.ScreenPointToRay(Mouse.current.position.ReadValue());
+#else
+        Ray ray = Camera.main.ScreenPointToRay(inputManager.touchControls.Touch.TouchPosition.ReadValue<Vector2>());
+#endif
+
+        if (Physics.Raycast(ray, out var hit))
+        {
+            EmptyIdentifyResults();
+            EmptyBuildingListResults();
+            SelectedResult = 0;
+            var arcGISRaycastHit = arcGISMapComponent.GetArcGISRaycastHit(hit);
+            var geoPosition = arcGISMapComponent.EngineToGeographic(hit.point);
+            var cameraGeoPosition = arcGISMapComponent.EngineToGeographic(Camera.main.transform.position);
+            var result = arcGISMapComponent.View.IdentifyLayersAsync(geoPosition, cameraGeoPosition, -1);
+            result.Wait();
+
+            if (!result.IsCanceled() && result.GetError() == null)
+            {
+                resultValue = result.Get();
+                ParseResults(SelectedResult, resultValue);
+                PopulateBuildingList();
+            }
+        }
+    }
+
+    public void ParseResults(ulong NumberOfResults, ArcGISImmutableCollection<ArcGISIdentifyLayerResult> ResultValue)
+    {
+        if (ResultValue.IsEmpty())
+        {
+            Debug.LogWarning("No Results Found");
+            return;
+        }
+
+        var elements = ResultValue.At(0).GeoElements;
+        resultsLength = elements.GetSize();
+
+        if (resultsLength == 0)
+        {
+            Debug.LogWarning("No Results Found");
+            return;
+        }
+        else if (resultsLength == 1)
+        {
+            DisableButtons(false);
+        }
+        else if (resultsLength > 1)
+        {
+
+            increaseResult.interactable = true;
+            decreaseResult.interactable = false;
+        }
+
+        totalNumberOfBuildingsText.text = $"total: {resultsLength}";
+        resultAmount.text = $"{SelectedResult + 1} of {resultsLength}";
+        var attributes = elements.At(NumberOfResults).Attributes;
+        var keys = attributes.Keys;
+        buildingsView.SetActive(false);
+        results.SetActive(true);
+
+        try
+        {
+            var id = attributes["OBJECTID"];
+            selectedID = Convert.ToInt32(id.ToString());
+            Shader.SetGlobalFloat("_SelectedObjectID", selectedID);
```

Comment:
I'd suggest adding a comment explaining this. There is no indication to the user that this is the part of the code making the shader "selection" happen.